viewing_party/party.py

```diff
@@ -1,23 +1,312 @@
+    user_watched_list = get_user_watched_list(user_data)
+    user_unique_list =[]
+
+    # for movie_dict in range(len(user_watched_list)):
+    #     if user_watched_list[movie_dict] not in friends_watched_list:
+    #         user_unique_list.append(user_watched_list[movie_dict])
+    # return user_unique_list
+    
+    #ENHANCE CODES FOR READABILITY
+    for movie_dict in user_watched_list:
+        if movie_dict not in friends_watched_list:
+            user_unique_list.append(movie_dict)
+    print(user_unique_list)
+    return user_unique_list
+
+def get_friends_unique_watched(user_data):
+    """Function to get a list of friends' unique movie dictionaries that user has not watched
+    Input: user data
+    Output: list of friends movie dicts that user has not watched
+    
+    """
+    friends_watched_list = get_friends_watched_list(user_data)
+    user_watched_list = get_user_watched_list(user_data)
+    friends_unique_list = []
+    
+    # for friends_dict in range(len(friends_watched_list)):
+    #     if friends_watched_list[friends_dict] not in user_watched_list:
+    #         friends_unique_list.append(friends_watched_list[friends_dict])
+    # return friends_unique_list
+    
+    #ENHANCE CODES FOR READABILITY
+    for friends_dict in friends_watched_list:
+        if friends_dict not in user_watched_list:
+            friends_unique_list.append(friends_dict)
+    # print(friends_unique_list)
+    return friends_unique_list
+
+
+def get_friends_watched_list(user_data):
+    """Helper function to create friends' watched list
+    Input: user_data
+    Output: list of friends watched movie dictionaries
+    """
+    friends_watched_list = []
+
+    if len(user_data["friends"]) == 0:
+        friends_watched_list = []
+    
+    else:
+        # for index1 in range(len(user_data["friends"])): #first loop to give us the movie list (of dictionaries) of "friends"
+        #     for index2 in range(len(user_data["friends"][index1]["watched"])): #second loop give us each element (list of dictionaries of "friends' watched list")
+        #         for index3 in range(len(user_data["friends"][index1]["watched"][index2])): #third loop give us access to each element within the list of friends watched list to compare title
+        #             if user_data["friends"][index1]["watched"][index2] not in friends_watched_list:
+        #                 friends_watched_list.append(user_data["friends"][index1]["watched"][index2])
+        
+        #ENHANCE CODES FOR READABILITY
+        for friend_movie_dicts in (user_data["friends"]): #first loop to give us the movie list (of dictionaries) of "friends"
+
+            for watched_dict in friend_movie_dicts["watched"]: #second loop give us each element (list of dictionaries of "friends' watched list")
+                for key,value in watched_dict.items(): #third loop give us access to each element within the list of friends watched list to compare title

```

Comment:
This second for loop is not necessary! You're not using the key or value that you pull from the dictionary. The code on lines 163 and 164 happen without it.